Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -474,14 +657,56 @@ class Gx2Fitter {
       ++result.surfaceCount;
       const GeometryIdentifier geoId = surface->geometryId();
       ACTS_DEBUG("Surface " << geoId << " detected.");
+
       const bool surfaceIsSensitive =
           (surface->associatedDetectorElement() != nullptr);
       const bool surfaceHasMaterial = (surface->surfaceMaterial() != nullptr);
-      const bool doMaterial = multipleScattering && surfaceHasMaterial;
+      // First we figure out, if we would need to look into material surfaces at
+      // all. Later, we also check, if the material slab is valid, otherwise we
+      // modify this flag to ignore the material completely.
+      bool doMaterial = multipleScattering && surfaceHasMaterial;
 
-      // Found material
+      // Found material - add a scatteringAngles entry if not done yet.
+      // Handling will happen later
       if (doMaterial) {
         ACTS_DEBUG("    The surface contains material, ...");
+
+        auto scatteringMapId = scatteringMap->find(geoId);
+        if (scatteringMapId == scatteringMap->end()) {
+          ACTS_DEBUG("    ... create entry in scattering map.");
+
+          detail::PointwiseMaterialInteraction interaction(surface, state,
+                                                           stepper);
+          // We need to evaluate the material to create the correct slab
+          const bool slabIsValid = interaction.evaluateMaterialSlab(
+              state, navigator, MaterialUpdateStage::FullUpdate);
+          double invSigma2 = 0.;
+          if (slabIsValid) {
+            const auto& particle =
+                parametersWithHypothesis->particleHypothesis();
+
+            const double sigma =
+                static_cast<double>(Acts::computeMultipleScatteringTheta0(
+                    interaction.slab, particle.absolutePdg(), particle.mass(),
+                    static_cast<float>(
+                        parametersWithHypothesis->parameters()[eBoundQOverP]),
+                    particle.absoluteCharge()));
+            ACTS_VERBOSE(
+                "        The Highland formula gives sigma = " << sigma);
+            invSigma2 = 1. / sigma / sigma;
```

Comment:
```suggestion
            invSigma2 = 1. / std::pow(sigma, 2);
```
`1 / a / a` looks a bit odd to me

Reply:
as far as I know and tested `std::pow(x, int)` will be optimized out. but maybe it is really just for 2